GitExtensions/Program.cs

```diff
@@ -139,11 +139,12 @@ private static string GetWorkingDir(string[] args)
             string workingDir = string.Empty;
             if (args.Length >= 3)
             {
-                if (Directory.Exists(args[2]))
-                    workingDir = GitModule.FindGitWorkingDir(args[2]);
+                var startDir = Path.GetFullPath(args[2]);
```

Comment:
By doing this you increase the risk of crashing the app if there is a problem with the supplied path...
https://msdn.microsoft.com/en-us/library/system.io.path.getfullpath(v=vs.110).aspx?f=255&mspperror=-2147217396#Anchor_1

It is probably more sensible to verify that the supplied path exists and only then perform the resolution.

Reply:
fixed